packages/flame/lib/src/extensions/matrix4.dart

```diff
@@ -78,6 +80,13 @@ extension Matrix4Extension on Matrix4 {
     return transform2(out);
   }
 
+  /// Transform [rect] of type [Rect] using the transformation defined by this.
+  Rect transformRect(Rect rect) {
```

Comment:
Maybe this live in a `RectExtension` and take a transform as the argument instead? I think that feels a bit more natural.

Reply:
Agreed! That makes more sense to me as well